src/pages/index.js

```diff
@@ -1,248 +1,248 @@
-});
+function createCard(data) {
+  return new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleCardDeleteClick,
+    handleCardLike,
+    handleCardLikeRemove
+  );
+}
 
-const contentCardPreview = new PopupWithImage({
-  popupSelector: ".js-preview-popup",
-});
+let cardSection;
 
-const cardSection = new Section({ items: cardData }, renderCard, "#section");
-cardSection.renderItems(cardData);
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section({ items: cards }, renderCard, "#section");
 
-const newUserInfo = new UserInfo("#profilename", "#profilejob");
+    function renderCard(cards) {
+      const cardElement = createCard(cards);
+      cardSection.addItem(cardElement.getView());
+    }
 
-function renderCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  // cardListEl.prepend(cardElement.getView());
-  cardSection.addItem(cardElement.getView());
-}
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
-function handleImageClick(cardData) {
-  console.log(cardData);
-  contentCardPreview.openModal(cardData);
-}
+const newCardPopup = new PopupWithForm({
+  popupSelector: "#contentModal",
+  handleFormSubmit: handleCardFormSubmit,
+});
 
-// ---- Profile Modal open and close event listeners -------------------------------------
+contentAddButton.addEventListener("click", () => {
+  cardFormValidator.toggleButtonState();
+  newCardPopup.openModal();
+});
 
-const editProfileButton = document.querySelector(
-  "#profile__edit-button-action"
+function handleCardFormSubmit(cardValues) {
+  const submitButton = document.getElementById("modalContentSubmit");
+  const originalButtonText = submitButton.textContent;
+  // Change button text to "Saving..."
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardValues)
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      cardSection.prependItems(newCard.getView());
+      newCardPopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = originalButtonText;
+    });
+}
+
+const newUserInfo = new UserInfo(
+  "#profilename",
+  "#profilejob",
+  "#profileImage"
 );
 
+api
+  .loadUserInfo()
+  .then((userData) => {
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 editProfileButton.addEventListener("click", () => {
-  // nameInput.value = profileName.textContent;
-  // jobInput.value = profileJob.textContent;
-  const data = newUserInfo.getUserInfo();
-  nameInput.value = data.name;
-  jobInput.value = data.job;
-  cardFormValidator.toggleButtonState();
+  nameInput.value = profileName.textContent;
+  jobInput.value = profileJob.textContent;
   profileCardPopup.openModal();
```

Comment:
`newUserInfo.getUserInfo();` should be used on profile opening like you had previously